server/src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -71,41 +71,55 @@
  * elected as master, it requests metaData from other master eligible nodes. After that, master node performs re-conciliation on the
  * gathered results, re-creates {@link ClusterState} and broadcasts this state to other nodes in the cluster.
  */
-public class GatewayMetaState implements ClusterStateApplier, CoordinationState.PersistedState {
+public class GatewayMetaState implements PersistedState {
     protected static final Logger logger = LogManager.getLogger(GatewayMetaState.class);
 
     private final MetaStateService metaStateService;
     private final Settings settings;
-    private final ClusterService clusterService;
-    private final TransportService transportService;
+    private final SetOnce<PersistedState> persistedState = new SetOnce<>();
```

Comment:
I think this field requires a comment, explaining why we can not just return this (which implements PersistedState).

Reply:
++ although I hope we can remove the `implements PersistedState` in the near future. See https://github.com/elastic/elasticsearch/pull/46326/commits/59c50d0afa2f2a7771e8dcb4489f2da42fade6cf

Reply:
Thanks, looks good